source/extensions/path/uri_template_lib/uri_template_internal.h

```diff
@@ -0,0 +1,138 @@
+#pragma once
+
+#include <optional>
+#include <string>
+#include <variant>
+#include <vector>
+
+#include "absl/container/flat_hash_set.h"
+#include "absl/status/status.h"
+#include "absl/status/statusor.h"
+#include "absl/strings/string_view.h"
+#include "absl/types/optional.h"
+#include "absl/types/variant.h"
+#include "re2/re2.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace UriTemplate {
+
+namespace Internal {
+
+/**
+ * String to be concatenated in rewritten url
+ */
+using Literal = absl::string_view;
+
+/**
+ * Determines what operations to use on the input pattern segment
+ */
+enum class Operator { PathGlob, TextGlob };
+
+/**
+ * Represents a pattern variable. Variables are included in both path match and rewrite paths.
+ */
+struct Variable {
+  absl::string_view var_name_;
+  std::vector<absl::variant<Operator, Literal>> var_match_;
+
+  Variable(absl::string_view name, std::vector<absl::variant<Operator, Literal>> match)
+      : var_name_(name), var_match_(match) {}
+
+  std::string debugString() const;
+};
+
+using ParsedSegment = absl::variant<Operator, Variable, Literal>;
+
+/**
+ * Represents the parsed path including literals and variables.
+ */
+struct ParsedUrlPattern {
```

Comment:
changed to be consistent

Reply:
It's just the path.

Reply:
Ah, ok. So here and elsewhere, let's be clear with naming name this is a path. So here `ParsedPathPattern` and elsewhere `path_pattern` and `path` instead of `url_pattern` and url`.

Reply:
changed